scripts/deprecate-npm-package.ts

```diff
@@ -0,0 +1,96 @@
+import execa from "execa";
+
+interface DeprecateOptions {
+  deprecateMessage: string;
+  packageNamesToBeDeprecated: string[];
+  versionsToBeDeprecated: RegExp;
+  isProduction?: boolean;
+}
+
+/**
+ * Options for deprecating npm packages by using this script
+ * `npm deprecate` (https://docs.npmjs.com/cli/v8/commands/npm-deprecate) supports using semver for specifying versions.
+ *  However, canary version cannot be deprecated easily.
+ *  e.g., `~1.0.0-0` will match:
+ *  - `1.0.0-xyz` (canary version)
+ *  - `1.0.x` (stable version)
+ *  but NOT 1.0.1-xyz (canary version)
+ *  Therefore, to match both versions below and you need a union version (~1.0.0-0 || ~1.0.1-0) or multiple commands:
+ *  - `1.0.0-xyz`
+ *  - `1.0.1-xyz`
+ *  We should not use multiple commands, as the server has certain limit.
+ */
+const deprecateOptions: DeprecateOptions = {
+  /**
+   * Deprecate message for all packages/versions to be deprecated.
+   */
+  deprecateMessage: "1.x is no longer maintained.",
+  /**
+   * Package names to be deprecated
+   */
+  packageNamesToBeDeprecated: [
+    "@sap-cloud-sdk/analytics",
+    "@sap-cloud-sdk/core",
+    "@sap-cloud-sdk/eslint-config",
+    "@sap-cloud-sdk/generator",
+    "@sap-cloud-sdk/generator-common",
+    "@sap-cloud-sdk/openapi-generator",
+    "@sap-cloud-sdk/test-util",
+    "@sap-cloud-sdk/util"
+  ],
+  /**
+   * Regex used as a filter, indicating which versions should be deprecated.
+   * The group is crucial for picking the stable version prefix, which cannot be removed.
+   * Check the doc of the `deprecateOptions` object for more details.
+   */
+  versionsToBeDeprecated: /(1\.\d+\.\d+).*/,
```

Comment:
[q] Wouldn't `/(1\.\d+\.\d+)/` be enough to match the version prefix part. How it's done currently is also good for me.

Reply:
Yes, you are right.
I will add `^` and therefore use `/^(1\.\d+\.\d+)/`;)